advanced/Scripts/api.sh

```diff
@@ -25,26 +25,48 @@ TestAPIAvailability() {
     ports="$(pihole-FTL --config webserver.port)"
     port="${ports%%,*}"
 
-    # if the port ends with an "s", it is a secure connection
-    if [ "${port#"${port%?}"}" = "s" ]; then
-        # remove the "s" from the port
-        API_PROT="https"
-        API_PORT="${port%?}"
-    elif [ "${port#"${port%?}"}" = "r" ]; then
-        # if the port ends in "r", it is a redirect
-        API_PROT="http"
-        # remove the "r" from the port
-        API_PORT="${port%?}"
-    else
-        API_PROT="http"
-        API_PORT="${port}"
-    fi
+    # Iterate over comma separated list of ports
+    while [ "${port}" != "${ports}" ]; do
```

Comment:
This won't allow to check the last set port, as in this case `"${port}" == "${ports}"`

See

```
++ pihole-FTL --config webserver.port
+ ports=80r,443s
+ port=80r
+ '[' 80r '!=' 80r,443s ']'
+ '[' r = s ']'
+ '[' r = r ']'
+ API_PORT=0
+ '[' '!' 0 = 0 ']'
+ ports=443s
+ port=443s
+ '[' 443s '!=' 443s ']'
+ '[' 0 = 0 ']'
+ echo 'API not available at: '

```